### PR TITLE
Automatic loom-js challenges

### PR DESCRIPTION
**Pull request by gakonst**
As title. Depends on https://github.com/loomnetwork/loom-js/pull/105. 

- Should change package.json to match latest release once https://github.com/loomnetwork/loom-js/pull/107 gets merged.

### CODE REVIEW

File: loom_js_test/src/challenge-after-demo.ts [hunk 1]
```diff
@@ -86,15 +87,18 @@ export async function runChallengeAfterDemo(t: test.Test) {
...
   // Having successufly challenged Mallory's exit Dan should be able to exit the coin
+  await sleep(2000)
   await dan.startExitAsync({
     slot: deposit1Slot,
     prevBlockNum: coin.depositBlockNum,
     exitBlockNum: plasmaBlock3
   })
-  //dan.stop_watching_exits(deposit1_utxo)
+
+  dan.stopWatching(deposit1Exit, console.log(`Stopped watching ${deposit1Slot}`))
```
Comment (enlight) [inline loom_js_test/src/challenge-after-demo.ts]: I've changed the signature for this function, so it returns a promise now, this should be rewritten like so:
```typescript
await dan.stopWatchingAsync(deposit1Exit)
console.log(`Stopped watching ${deposit1Slot}`)
```

File: loom_js_test/src/challenge-after-demo.ts [hunk 2]
```diff
@@ -86,15 +87,18 @@ export async function runChallengeAfterDemo(t: test.Test) {
 
   // Mallory's exit should be auto-challenged by Dan's client, but watching/auto-challenge hasn't
   // been implemented yet, so challenge the exit manually for now...
-  await dan.challengeAfterAsync({ slot: deposit1Slot, challengingBlockNum: plasmaBlock3 })
+  // console.log('DAN WOULD CHALLENGE WITH', plasmaBlock3)
+  // await dan.challengeAfterAsync({ slot: deposit1Slot, challengingBlockNum: plasmaBlock3 })
```
Comment (enlight) [inline loom_js_test/src/challenge-after-demo.ts]: Lets remove all this commented out code if the watching is working now.

File: loom_js_test/package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
   "dependencies": {
     "bn.js": "^4.11.8",
     "ethereumjs-util": "^5.2.0",
-    "loom-js": "^1.17.0",
```
Comment (mattkanwisher) [inline loom_js_test/package.json]: looms do an npm publish before we merge this branch

**Comment by mattkanwisher** [comment]
largely looks good, I'll let @enlight take a quick sanity check also. just want to get the npm published first

**Pull request merged**